internal/core/src/segcore/segment_c.cpp

```diff
@@ -100,6 +101,7 @@ Retrieve(CSegmentInterface c_segment, CRetrievePlan c_plan, uint64_t timestamp,
         result->proto_size = size;
         return milvus::SuccessCStatus();
     } catch (std::exception& e) {
+        result = nullptr;
```

Comment:
Doesn't it need to manually destruct the variables pointed to by pointer members in CRetrieveResult?